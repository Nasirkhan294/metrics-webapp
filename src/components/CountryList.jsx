import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { BiRightArrowAlt } from 'react-icons/bi';
import { getCountries } from './redux/country/countrySlice';
import styles from '../styles/CountryList.module.css';

const CountryList = () => {
  const { countries, isLoading, error } = useSelector((state) => state.countries);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCountries());
  }, []);

  if (isLoading) {
    return <p>Loading ...</p>;
  }

  if (error) {
    return `Error ${error}`;
  }
  return (
    <div className={styles.container}>
      {countries.map((country) => (
        <div key={country.name} className={styles.countryContainer}>
          <div className={styles.thumbnailContainer}>
            <img src={country.flag} alt={country.demonym} className={styles.thumbnail} />
          </div>
          <div className={styles.details}>
            <h2>{country.name}</h2>
            <p>{country.region}</p>
            <p>{country.population}</p>
          </div>
          <BiRightArrowAlt className={styles.arrow} />
        </div>
      ))}
    </div>
  );
};

export default CountryList;
