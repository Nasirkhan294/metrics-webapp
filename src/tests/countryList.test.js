import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import CountryList from '../components/CountryList';

const mockStore = configureStore([thunk]);

describe('CountryList component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      countries: {
        isLoading: false,
        error: null,
        countries: [
          {
            name: 'Australia',
            population: 25600000,
            flag: 'https://restcountries.com/data/aus.svg',
            region: 'Oceania',
          },
          {
            name: 'Brazil',
            population: 213000000,
            flag: 'https://restcountries.com/data/bra.svg',
            region: 'Americas',
          },
          {
            name: 'Canada',
            population: 37742154,
            flag: 'https://restcountries.com/data/can.svg',
            region: 'Americas',
          },
        ],
      },
    });
  });

  it('should render the CountryList component correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <CountryList />
        </MemoryRouter>
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
