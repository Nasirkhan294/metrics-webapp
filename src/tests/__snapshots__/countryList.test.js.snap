// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryList component should render the CountryList component correctly 1`] = `
<DocumentFragment>
  <div
    class="banner"
  >
    <input
      class="input"
      placeholder="search country ..."
      type="text"
      value=""
    />
  </div>
  <div
    class="container"
  >
    <div
      class="countryContainer"
    >
      <div
        class="thumbnailContainer"
      >
        <img
          alt="Australia"
          class="thumbnail"
          src="https://restcountries.com/data/aus.svg"
        />
      </div>
      <div
        class="details"
      >
        <h2>
          Australia
        </h2>
        <p>
          Oceania
        </p>
        <p>
          25600000
        </p>
      </div>
      <a
        href="/country/Australia"
      >
        <svg
          class="arrow"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m11.293 17.293 1.414 1.414L19.414 12l-6.707-6.707-1.414 1.414L15.586 11H6v2h9.586z"
          />
        </svg>
      </a>
    </div>
    <div
      class="countryContainer"
    >
      <div
        class="thumbnailContainer"
      >
        <img
          alt="Brazil"
          class="thumbnail"
          src="https://restcountries.com/data/bra.svg"
        />
      </div>
      <div
        class="details"
      >
        <h2>
          Brazil
        </h2>
        <p>
          Americas
        </p>
        <p>
          213000000
        </p>
      </div>
      <a
        href="/country/Brazil"
      >
        <svg
          class="arrow"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m11.293 17.293 1.414 1.414L19.414 12l-6.707-6.707-1.414 1.414L15.586 11H6v2h9.586z"
          />
        </svg>
      </a>
    </div>
    <div
      class="countryContainer"
    >
      <div
        class="thumbnailContainer"
      >
        <img
          alt="Canada"
          class="thumbnail"
          src="https://restcountries.com/data/can.svg"
        />
      </div>
      <div
        class="details"
      >
        <h2>
          Canada
        </h2>
        <p>
          Americas
        </p>
        <p>
          37742154
        </p>
      </div>
      <a
        href="/country/Canada"
      >
        <svg
          class="arrow"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m11.293 17.293 1.414 1.414L19.414 12l-6.707-6.707-1.414 1.414L15.586 11H6v2h9.586z"
          />
        </svg>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
